quadriga/src/main/java/edu/asu/diging/quadriga/web/DisplayCollectionController.java

```diff
@@ -22,34 +49,55 @@ public class DisplayCollectionController {
 
     @Autowired
     private EventGraphService eventGraphService;
-        
+    
     private Logger logger = LoggerFactory.getLogger(getClass());
 
-    @RequestMapping(value = "/auth/collections/{id}", method = RequestMethod.GET)
-    public String get(@PathVariable String id, Model model) {
-        
+    @RequestMapping(value = "/auth/collections/{collectionId}", method = RequestMethod.GET)
+    public String get( @RequestParam(required = false) Integer page , @RequestParam(required = false) Integer  size, @PathVariable String collectionId, Model model) {
+
+        // Get collection details
         Collection collection;
         try {
-            collection = collectionManager.findCollection(id);
+            collection = collectionManager.findCollection(collectionId);
             if(collection == null) {
-            	logger.error("Couldn't find collection: ", id);
+            	logger.error("Couldn't find collection: ", collectionId);
                 return "error404Page";
             }
         } catch (InvalidObjectIdException e) {
             logger.error("Couldn't find collection ", e);
             return "error404Page";
         }
-        
+
+        model.addAttribute("collection", collection);
+
+//      Determine page number and size for network pagination      
+        page = (page == null || page < 0) ? 0 : page - 1;
+        size = (size == null || size < 1) ? 10 : size;
```

Comment:
instead of checking for null you can just specify a default value in the RequestParam annotation